src/tests/integration/MetaMorpho.t.sol

```diff
@@ -0,0 +1,300 @@
+        
+        assertEq(plan.targetAssets(), d3mDeposit);
+        
+        assertEq(morpho.market(marketParams.id()).totalSupplyAssets, marketSupplyBefore + depositedAssets);
+        
+        assertEq(spDai.balanceOf(address(pool)), depositedAssets);
+        assertEq(spDai.totalSupply(), spDaiTotalSupplyBefore + depositedAssets);
+        assertEq(spDai.totalAssets(), spDaiTotalAssetsBefore + morpho.expectedSupplyAssets(marketParams, address(spDai)));
+        assertEq(spDai.maxDeposit(address(pool)), spDaiMaxDepositBefore - depositedAssets);
+        
+        assertEq(dai.balanceOf(address(morpho)), morphoBalanceBefore + depositedAssets);
+        assertEq(dai.totalSupply(), daiTotalSupplyBefore + depositedAssets);
+    }
+
+    function testDepositInCappedMarket(uint184 cap, uint256 d3mDeposit) public {
+        uint256 marketSupplyBefore = morpho.market(marketParams.id()).totalSupplyAssets;
+
+        setCap(marketParams, cap);
+        
+        // Set target assets at `d3mDeposit` and exec.
+        vm.prank(operator);
+        plan.setTargetAssets(d3mDeposit);
+        hub.exec(ilk);
+
+        (,,, uint256 line,) = VatAbstract(vat).ilks(ilk);
+        uint256 depositedAssets = min(d3mDeposit, min(line * 1e18 / 1e45, cap));
+        
+        assertEq(morpho.market(marketParams.id()).totalSupplyAssets, marketSupplyBefore + depositedAssets);
+    }
+
+    function testWithdrawLiquid(uint256 target1, uint256 target2) public {
+        uint256 marketSupplyStart = morpho.market(marketParams.id()).totalSupplyAssets;
+        
+        (,,, uint256 line,) = VatAbstract(vat).ilks(ilk);
+        vm.assume(target1 < line * 1e18 / 1e45);
+        
+        adjustDebt(int256(target1));
+        
+        vm.assume(target2 <= pool.assetBalance());
+        
+        uint256 marketSupplyMiddle = morpho.market(marketParams.id()).totalSupplyAssets;
+        
+        uint256 depositedAssets1 = min(target1, line * 1e18 / 1e45);
+        uint256 expectedWithdraw = min(pool.maxWithdraw(), pool.assetBalance() - target2);
+        uint256 depositedAssets2 = depositedAssets1 - expectedWithdraw;
+
+        // Set target assets at `target2` and exec.
+        vm.prank(operator);
+        plan.setTargetAssets(target2);
+        hub.exec(ilk);
+
+        uint256 marketSupplyEnd = morpho.market(marketParams.id()).totalSupplyAssets;
+        
+        assertGe(marketSupplyMiddle, marketSupplyStart);
+        assertLe(marketSupplyEnd, marketSupplyMiddle);
+        assertEq(marketSupplyEnd, marketSupplyMiddle - expectedWithdraw);
+        assertEq(marketSupplyEnd, marketSupplyStart + depositedAssets2);
+    }
+
+    function testWithdrawIlliquid(uint256 target1, uint256 target2, uint256 borrow) public {
+        uint256 marketSupplyStart = morpho.market(marketParams.id()).totalSupplyAssets;
+        uint256 marketBorrowStart = morpho.market(marketParams.id()).totalBorrowAssets;
+        vm.assume(borrow < marketSupplyStart - marketBorrowStart);
```

Comment:
Replace these with `_bound()`. `vm.assume()` is too slow with lots of fuzz runs.